packages/libs/coreui/src/components/inputs/SelectList.tsx

```diff
@@ -30,31 +37,36 @@ export default function SelectList<T extends string>({
   isDisabled = false,
   isLoading = false,
   instantUpdate = false,
+  transitionDuration,
   ...props
 }: SelectListProps<T>) {
   const [selected, setSelected] = useState<SelectListProps<T>['value']>(value);
   const [buttonDisplayContent, setButtonDisplayContent] = useState<ReactNode>(
     getDisplayContent(value, items, defaultButtonDisplayContent)
   );
+  const [isOpen, setIsOpen] = useState(false);
 
-  const onClose = () => {
-    onChange(selected);
+  const onClose = useCallback(() => {
+    if (!instantUpdate) onChange(selected);
```

Comment:
Added this to be symmetrical with `handleCheckboxListUpdate`. No need to update on closing if instant updates were already made. This was probably a harmless oversight.